### ISSUE TITLE
Pybind support

### ISSUE THREAD

**Issue by larryliu0820** [pytorch/executorch#1006]
After enabling pybind we should be able to migrate a lot more internal tests to OSS. Currently the pybind repo is checked in third-party/ but need some wiring to make it work for python_cpp_extension BUCK macros.

**Comment by larryliu0820** [comment]
Existing attempt: https://github.com/pytorch/executorch/tree/pybind

**Comment by mergennachin** [comment]
Hi @larryliu0820 

Is (https://github.com/pytorch/executorch/pull/332/) sufficient 

Can you mention about what's the status, how people can use portable_lib pybindings, add documentation (if necessary) and close this task?

**Comment by larryliu0820** [comment]
#332 enabled building pybind shared library and used it in end2end test. This is a good progress but we need some UX work:

1. Package the pybind shared library along into pip install. This exposes python modules under executorch.extension.pybindings for users to use.
2. Hook this up with selective build so we can do selective build based on .pte model files.

**Comment by mergennachin** [comment]
> Hook this up with selective build so we can do selective build based on .pte model files.

Was there an actual need from developers? Why is this important? 

As far as I can tell, pybind is mostly used for testing as well as fast iteration and experimentation.

I don't expect developers to use pybind for an actual on-device deployment, and hence, selective build for pybind doesn't seem necessary. It seems enough to just link against all portable operators.

**Comment by larryliu0820** [comment]
Yeah not saying we are using pybind for actual model development. Pybind exposes a python API to extracts operator information from model and we would want users to use that API. Currently users can only hardcode operator names in OSS.

**Comment by mergennachin** [comment]
Ah, yes yes. I remember now. 

Thanks a lot, this is great progress so far

**Comment by swolchok** [comment]
@larryliu0820 can we close this? we use pybind, not sure if there's more to this issue
